lib/discordrb/bot.rb

```diff
@@ -548,7 +548,11 @@ def update_status(status, activity, url, since = 0, afk = false, activity_type =
       @streamurl = url
       type = url ? 1 : activity_type
 
-      activity_obj = activity || url ? { 'name' => activity, 'url' => url, 'type' => type } : nil
+      activity_obj = if activity_type == 4
+                       { 'name' => activity, 'type' => type, 'state' => activity }
```

Comment:
`type` here will (or should) always be 4. If something is changed in the logic above and `type` ends up not 4 then this might cause unwanted behaviour.

Setting `'type' => 4` or `'type' => activity_type` might be better.

Reply:
The only problem I can see is that I used `activity_type == 4` instead of `type == 4` by mistake, which could have caused inconsistency in some cases.

But there's no reason to hard-code the value instead of using that of the variable, as long as it's the right one that's used and verified by the condition

I force push the change